mne/time_frequency/tfr.py

```diff
@@ -704,16 +708,72 @@ def plot(self, picks=None, baseline=None, mode='mean', tmin=None,
+                        colorbar=colorbar, picker=False, cmap=cmap)
             if title:
                 fig.suptitle(title)
+            colorbar = False  # only one colorbar for multiple axes
         if show:
             plt.show()
         return fig
 
+    def _onselect(self, eclick, erelease, baseline, mode, vmin, vmax, layout,
+                  cmap):
+        """Callback function called by rubber band selector in channel tfr."""
+        import matplotlib.pyplot as plt
+        from ..viz import plot_tfr_topomap
+        if abs(eclick.x - erelease.x) < .1 or abs(eclick.y - erelease.y) < .1:
+            return
+        plt.ion()  # turn interactive mode on
+        tmin = round(min(eclick.xdata, erelease.xdata) / 1000., 1)  # ms to s
+        tmax = round(max(eclick.xdata, erelease.xdata) / 1000., 1)
+        fmin = round(min(eclick.ydata, erelease.ydata), 1)  # Hz
+        fmax = round(max(eclick.ydata, erelease.ydata), 1)
+        types = list()
+        eegs = pick_types(self.info, meg=False, eeg=True, ref_meg=False)
+        if len(eegs) > 0:
+            types.append('eeg')
+        if len(pick_types(self.info, meg='mag', ref_meg=False)) > 0:
+            types.append('mag')
+        if len(pick_types(self.info, meg='grad', ref_meg=False)) > 0:
+            types.append('grad')
+        fig, axis = plt.subplots(1, len(types))
+        if isinstance(axis, plt.Axes):
+            axis = [axis]
+        fig.suptitle('%s s - %s s, %s Hz - %s Hz' % (tmin, tmax, fmin, fmax),
+                     y=0.04)
+        plot_idx = 0
+        try:
+            if 'grad' in types:
+                plot_tfr_topomap(self, ch_type='grad', tmin=tmin, tmax=tmax,
+                                 fmin=fmin, fmax=fmax, layout=layout,
+                                 baseline=baseline, mode=mode, cmap=cmap,
+                                 title='grad', vmin=vmin, vmax=vmax,
+                                 axes=axis[plot_idx])
+                plot_idx += 1
+            if 'mag' in types:
+                plot_tfr_topomap(self, ch_type='mag', tmin=tmin, tmax=tmax,
+                                 fmin=fmin, fmax=fmax, layout=layout,
+                                 baseline=baseline, mode=mode, cmap=cmap,
+                                 title='mag', vmin=vmin, vmax=vmax,
+                                 axes=axis[plot_idx])
+                plot_idx += 1
+            if 'eeg' in types:
+                plot_tfr_topomap(self, ch_type='eeg', tmin=tmin, tmax=tmax,
+                                 fmin=fmin, fmax=fmax, layout=layout,
+                                 baseline=baseline, mode=mode, cmap=cmap,
+                                 title='eeg', vmin=vmin, vmax=vmax,
+                                 axes=axis[plot_idx])
+        except IndexError:  # if the user selects a window that is too small
+            logger.info('The selected area is too small. '
+                        'Select a larger time-frequency window.')
```

Comment:
Any chances we can avoid the try/except and check the input data instead? Is this error predictable?